From 7967b6eb4ec78f5553a02eb97236ef3b91d80998 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 30 Aug 2011 14:21:49 +0000
Subject: xsettings: Handle rgba-order key

Seeing as we have a key to set that up, let's hook it up.

https://bugzilla.gnome.org/show_bug.cgi?id=657525
---
diff --git a/plugins/xsettings/gsd-xsettings-manager.c b/plugins/xsettings/gsd-xsettings-manager.c
index c5980f8..645d3c7 100644
--- a/plugins/xsettings/gsd-xsettings-manager.c
+++ b/plugins/xsettings/gsd-xsettings-manager.c
@@ -246,10 +246,12 @@ xft_settings_get (GnomeXSettingsManager *manager,
 {
         GsdFontAntialiasingMode antialiasing;
         GsdFontHinting hinting;
+        GsdFontRgbaOrder order;
         gboolean use_rgba = FALSE;
 
         antialiasing = g_settings_get_enum (manager->priv->plugin_settings, FONT_ANTIALIASING_KEY);
         hinting = g_settings_get_enum (manager->priv->plugin_settings, FONT_HINTING_KEY);
+        order = g_settings_get_enum (manager->priv->plugin_settings, FONT_RGBA_ORDER_KEY);
 
         settings->antialias = (antialiasing != GSD_FONT_ANTIALIASING_MODE_NONE);
         settings->hinting = (hinting != GSD_FONT_HINTING_NONE);
@@ -272,6 +274,24 @@ xft_settings_get (GnomeXSettingsManager *manager,
                 break;
         }
 
+        switch (order) {
+        case GSD_FONT_RGBA_ORDER_RGBA:
+                settings->rgba = "rgba";
+                break;
+        case GSD_FONT_RGBA_ORDER_RGB:
+                settings->rgba = "rgb";
+                break;
+        case GSD_FONT_RGBA_ORDER_BGR:
+                settings->rgba = "bgr";
+                break;
+        case GSD_FONT_RGBA_ORDER_VRGB:
+                settings->rgba = "vrgb";
+                break;
+        case GSD_FONT_RGBA_ORDER_VBGR:
+                settings->rgba = "vbgr";
+                break;
+        }
+
         switch (antialiasing) {
         case GSD_FONT_ANTIALIASING_MODE_NONE:
                 settings->antialias = 0;
--
cgit v0.9.0.2

